test/integration/test_utils.mjs

```diff
@@ -52,11 +53,37 @@ function loadAndWait(filename, selector, zoom, pageSetup, options) {
         global.integrationBaseUrl
       }?file=/test/pdfs/${filename}#zoom=${zoom ?? "page-fit"}${app_options}`;
 
-      await page.goto(url);
-      if (pageSetup) {
-        await pageSetup(page);
+      if (setups) {
+        await setups.prePageSetup?.(page);
+        if (setups.eventBusSetup) {
+          await page.evaluateOnNewDocument(setup => {
+            let app;
+            let eventBus;
+            Object.defineProperty(window, "PDFViewerApplication", {
```

Comment:
Could you explain a bit how this works exactly? I'd expect that if we load PDF.js that we set `PDFViewerApplication` to a new object, so wouldn't we then just overwrite the stubs we have here and effectively get a new event bus without the events we registered here?

Reply:
Oh, I think I understand how this works now. Before we load PDF.js we define `window.PDFViewerApplication` to this getter/setter pair, and when PDF.js loads and sets `window.PDFViewerApplication` to the real version in the line shown above that doesn't override this getter/setter pair but instead triggers its `set` method, and the same holds when `window.PDFViewerApplication.eventBus` is set. So if I'm not mistaken this more-or-less "hijacks" setter actions to run our test logic in addition to just setting the value? If so, it might be good to shortly explain that in an inline comment here, but that is pretty clever.